flask_security/core.py

```diff
@@ -461,30 +461,52 @@ class Security(object):
 
     :param app: The application.
     :param datastore: An instance of a user datastore.
+    :param register_blueprint: to register the Security blueprint or not.
+    :param login_form: set form for the login view
+    :param register_form: set form for the register view
+    :param confirm_register_form: set form for the confirm register view
+    :param forgot_password_form: set form for the forgot password view
+    :param reset_password_form: set form for the reset password view
+    :param change_password_form: set form for the change password view
+    :param send_confirmation_form: set form for the send confirmation view
+    :param passwordless_login_form: set form for the passwordless login view
+    :param anonymous_user: class to use for anonymous user
     """
 
-    def __init__(self, app=None, datastore=None, **kwargs):
+    def __init__(self, app=None, datastore=None, register_blueprint=True,
+                 **kwargs):
         self.app = app
-        self.datastore = datastore
-
+        self._datastore = datastore
```

Comment:
I prefixed this attribute with an underscore so that both test cases pass whether datastore was passed to `__init__` or `init_app`. The issue with it being named `datastore` is that that takes precedence over calls to `Security.__getattr__()`:

```python
def test_access_datastore_from_factory(app, datastore):
    security = Security()  # sets Security.datastore to None
    security.init_app(app, datastore)  # sets Security._state['datastore'] to datastore

    assert security.datastore is not None  # fails, because this retrieves Security.datastore, not Security.__getattr__('datastore'), which would return Security._state['datastore']
```